### ISSUE TITLE
git tag latest version

### ISSUE THREAD

**Issue by jimmyl02** [moznion/go-optional#41]
please tag the latest commit with a git tag!

we were debugging an extremely difficult case with the sql driver valuer interface. the example error message was `sql: converting argument $5 type: incompatible data type for SQL driver Valuer on Option[T]`

after much time, we root caused the issue as `go get -u github.com/moznion/go-optional` installing the package from the `v0.11.0` tag which is much outdated past the latest commit. this meant the change to fix the sql driver in latest was not in the package we installed as a dependency!

for others facing this problem; you can get the latest version by using the git tag as `go get -u github.com/moznion/go-optional@6881072e44c133dd5708cff0301f4936a98dd087` which will use the latest ref and includes the sql interface fixes / latest version. I believe it would be a great help to the community to create a tag for the latest version!

**Comment by jimmyl02** [comment]
nevermind, it seems like main is broken with the following error
```
relocation target database/sql.convertAssign not defined
```

**Comment by levikline** [comment]
> nevermind, it seems like main is broken with the following error
> 
> ```
> relocation target database/sql.convertAssign not defined
> ```

Appears to be an issue with the scanner implementation:
```
//go:linkname sqlConvertAssign database/sql.convertAssign
func sqlConvertAssign(dest, src any) error
```

**Comment by moznion** [comment]
I'm sorry for my laziness.

I've just published the latest version [v0.12.0](https://github.com/moznion/go-optional/releases/tag/v0.12.0) that includes #32, #33, and #40 so it begins supporting the any type value which supported by `database/sql` on `Scan` and `Value` function.

Thank you for your patience.

**Comment by jimmyl02** [comment]
not at all! thanks for the great library and initiative, it's been an immense help for us
